src/visualize/VoronoiGeometries.cpp

```diff
@@ -147,20 +147,130 @@ VoronoiGeometry NuTo::Visualize::VoronoiGeometryBrick(int num, Spacing s)
     }
 }
 
-VoronoiGeometry NuTo::Visualize::VoronoiGeometryTriangle3Ip()
+/* =========================================================
+ *              TRIANGLE STUFF
+ * =========================================================
+ */
+
+#include <boost/polygon/voronoi.hpp>
+#include <Eigen/Dense> // for Eigen::inverse()
+
+typedef boost::polygon::point_data<double> VoronoiPoint;
+
+int AddUnique(std::vector<Eigen::VectorXd>* rExisingPoints, Eigen::Vector2d newPoint)
 {
-    VoronoiGeometry geometry;
-    geometry.pointCoordinates.push_back(Eigen::Vector2d(0, 0));
-    geometry.pointCoordinates.push_back(Eigen::Vector2d(0.5, 0));
-    geometry.pointCoordinates.push_back(Eigen::Vector2d(1, 0));
-    geometry.pointCoordinates.push_back(Eigen::Vector2d(0, 0.5));
-    geometry.pointCoordinates.push_back(Eigen::Vector2d(1. / 3., 1. / 3.));
-    geometry.pointCoordinates.push_back(Eigen::Vector2d(0.5, 0.5));
-    geometry.pointCoordinates.push_back(Eigen::Vector2d(0, 1));
-
-    geometry.voronoiCells.push_back({{0, 1, 4, 3}, NuTo::eCellTypes::QUAD});
-    geometry.voronoiCells.push_back({{1, 2, 5, 4}, NuTo::eCellTypes::QUAD});
-    geometry.voronoiCells.push_back({{4, 5, 6, 3}, NuTo::eCellTypes::QUAD});
+    for (size_t i = 0; i < rExisingPoints->size(); ++i)
+        if ((newPoint - (*rExisingPoints)[i]).squaredNorm() < 1.e-4)
+            return i;
+    rExisingPoints->push_back(newPoint);
+    return rExisingPoints->size() - 1;
+}
+
+std::vector<Eigen::Vector2d> TransformedHull(std::vector<Eigen::Vector2d> hullpoints, Eigen::Matrix2d T)
+{
+    std::vector<Eigen::Vector2d> hull;
+    for (Eigen::Vector2d p : hullpoints)
+        hull.push_back(T * p);
+    return hull;
+}
+
+std::vector<VoronoiPoint> TransformedIps(const NuTo::IntegrationTypeBase& integrationType, Eigen::Matrix2d T)
+{
+    std::vector<VoronoiPoint> points;
+    for (int i = 0; i < integrationType.GetNumIntegrationPoints(); ++i)
+    {
+        Eigen::Vector2d ip = T * Eigen::Vector2d(integrationType.GetLocalIntegrationPointCoordinates(i).x(),
+                                                 integrationType.GetLocalIntegrationPointCoordinates(i).y());
+        points.push_back({ip.x(), ip.y()});
+    }
+    return points;
+}
+
+std::vector<VoronoiPoint> ReflectPointsOnHull(std::vector<VoronoiPoint> points, std::vector<Eigen::Vector2d> hull)
+{
```

Comment:
I think this is a bit over the top - but since it works... It would probably be enough to add 3 points far enough outside the triangle to have the same effect.

Reply:
Previously, I calculated an intersection of the voronoi diagram with the hull. I removed that.
Now we modify points (by this reflection) in a way the resulting voronoi diagram _must have_ edges on the hull.